signer/src/ecdsa.rs

```diff
@@ -23,31 +23,43 @@
 //!
 //! struct SignableStr(&'static str);
 //!
-//! // Implementing `wsts::net::Signable` unlocks the signing functionality in this module.
-//! impl wsts::net::Signable for SignableStr {
-//!     fn hash(&self, hasher: &mut sha2::Sha256) {
-//!         hasher.update(self.0)
+//! // Implementing `ProtoSerializable` and conversion traits unlock the signing
+//! // functionality in this module.
+//! #[allow(clippy::derive_partial_eq_without_eq)]
+//! #[derive(Clone, PartialEq, ::prost::Message)]
+//! pub struct ProtoSignableStr {
+//!     /// The string
+//!     #[prost(string, tag = "1")]
+//!     pub string: ::prost::alloc::string::String,
+//! }
+//!
+//! impl ProtoSerializable for SignableStr {
+//!     type Message = ProtoSignableStr;
+//!
+//!     fn type_tag(&self) -> &'static str {
+//!         "SBTC_SIGNABLE_STR"
 //!     }
 //! }
 //!
 //! let msg = SignableStr("Sign me please!");
 //! let private_key = PrivateKey::try_from(&p256k1::scalar::Scalar::from(1337)).unwrap();
 //!
 //! // Sign the message.
-//! let signed_msg = msg
-//!     .sign_ecdsa(&private_key)
-//!     .expect("Failed to sign message");
+//! let signed_msg = msg.sign_ecdsa(&private_key);
 //!
 //! // Verify the signed message.
-//! assert!(signed_msg.verify());
+//! assert_eq!(signed_msg.recover_ecdsa(), Ok(private_key.into()));
```

Comment:
nit: is this last comment correct?

Reply:
Nope, fixing them now.

Reply:
Fixed in https://github.com/stacks-network/sbtc/pull/957/commits/304999adfa97b97d2ad69909203cf65b29132821.